pdm/installers/synchronizers.py

```diff
@@ -78,7 +81,11 @@ def __init__(
         self.ui = environment.project.core.ui
 
     @contextlib.contextmanager
-    def create_executor(self):
+    def create_executor(
+        self,
+    ) -> Iterator[
+        Union[Iterator, Iterator[ThreadPoolExecutor], Iterator[DummyExecutor]]
```

Comment:
Doesn't make sense, should be
`Union[Iterator[ThreadPoolExecutor], Iterator[DummyExecutor]]`

Reply:
Fixed here

[`312287c` (#354)](https://github.com/pdm-project/pdm/pull/354/commits/312287c375db739caa180500afbc976002515113#diff-3c7dbb09445dee162dcefb5ac080ace1d0becda785d2c3fb9cab128e5d94632cR86)